exonum-java-binding/core/rust/src/testkit/mod.rs

```diff
@@ -288,3 +291,21 @@ fn get_field_as_string(env: &JNIEnv, obj: JObject, field_name: &str) -> JniResul
         env.get_field(obj, field_name, "Ljava/lang/String;")?.l()?,
     )
 }
+
+/// DTO for time oracle service instantiation parameters.
+struct TimeServiceInstanceParams {
+    pub factory: TimeServiceFactory,
+    pub service_id: u32,
+    pub service_name: String,
+}
+
+impl<'a> Into<InstanceInitParams> for &'a TimeServiceInstanceParams {
```

Comment:
Remember the question from @dmitry-timofeev 

Implementing `Into` for reference is not that unusual, and is totally valid in situations like ours. It allows to use `TestKitBuilder::with_instance(Into<InstanceInitParams>)` without calling conversion method explicitly.